### ISSUE TITLE
HardClip reads in minimap2 aligned reads

### ISSUE THREAD

**Issue by qiangfu-eu** [nanoporetech/modkit#11]
I have a question about how modkit handling HardClip reads in minimap2. The issue is when hardclip is happening, part of the read sequences are removed. However, when I check MM and ML tag, it does not seems to be updated by minimap2 accordingly. Then there is a mismatch between what specified in MM/ML and what is in the remained read seqeuences.

How does modkit handling this?

I briefly checked my bam file. It seems all hardclip reads are of `supplementary alignment`. I am not sure how general this is the case. Dose minimap mark hardcipped reads as supplementary?

Many thanks for your help.

**Comment by ArtRand** [comment]
You're correct, without properly adjusting the MM/ML tags the tags cannot be decoded correctly when there are hard-clips. As of version 0.1.5, there is no special treatment of hard-clipped reads, but it's likely that the MM/ML tags will fail to be parsed correctly. When the MM/ML tags fail parsing, the read is discarded and the error should be logged. The issue you raise here (consistent handling and logging of reads that fail to be parsed or used) is similar to what you mention at the end of #10 could you either wrap the whole thing into a new issue or re-use this issue for that general problem? I think it's reasonable to have this fixed in the next release.

**Comment by qiangfu-eu** [comment]
I had this issue that when I run `modkit` on our BAM files. 

Due to the porechop step we run to remove adapter, many read sequences are trimmed. However, the corresponding MM/ML tags carried over from dorado called reads are not adapted accordingly, and simply transfered into minimap2 aligned BAM file.  These results in a lot of discrapency between sequence aligned and the methylation calling stored in MM/ML tags. When run `modit pileup` on BAM file, those reads will be skipped due to 'malformed MM delta list'. However, if a log file is not specified (`--log-filepath`), there is no error message or warning message to notify user about those skipped reads.

I noticed that some runing information are output in the stderr. It might be of great help to also give a warning of number of reads skipped or with issues and output this in the stderr. User can then decide by themself whether to check further with the problemetic reads.

Many thanks for developing a great tool.

**Comment by ArtRand** [comment]
@qiangfu-eu if you have a moment, could you try the latest released version on your modBAM containing hard-clips? It should output the number of skipped and failed reads, which would make debugging easier.

**Comment by ArtRand** [comment]
@ArtRand I'm going to close this due to inactivity. Let me know if you have any other questions.
